src/main/plugin-metadata/plugin-security.policy

```diff
@@ -1,5 +1,8 @@
 grant {
     permission java.lang.RuntimePermission "loadLibrary.opensearchknn_nmslib";
     permission java.lang.RuntimePermission "loadLibrary.opensearchknn_faiss";
+    permission java.lang.RuntimePermission "loadLibrary.opensearchknn_faiss_avx2";
     permission java.net.SocketPermission "*", "connect,resolve";
+    permission java.lang.RuntimePermission "accessDeclaredMembers";
```

Comment:
@navneet1v Could you take a quick look at changes in security policy? I think this is fine but want just a quick double check

Reply:
Checking.

Reply:
@naveentatikonda why we need this permission?

Reply:
We are adding this runtime permission for [oshi library](https://github.com/oshi/oshi/blob/master/oshi-core/src/main/java/oshi/util/platform/mac/SysctlUtil.java#L90-L106) to fetch sysctl cpu flags info on mac OS which is restricted by OpenSearch.
```
java.lang.IllegalArgumentException: Can't create an instance of class oshi.jna.ByRef$CloseableSizeTByReference, requires a public no-arg constructor: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "accessDeclaredMembers")
        at com.sun.jna.Klass.newInstance(Klass.java:68) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at com.sun.jna.NativeMappedConverter.defaultValue(NativeMappedConverter.java:65) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at com.sun.jna.NativeMappedConverter.<init>(NativeMappedConverter.java:56) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at com.sun.jna.NativeMappedConverter.getInstance(NativeMappedConverter.java:45) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at com.sun.jna.Function.convertArgument(Function.java:509) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at com.sun.jna.Function.invoke(Function.java:345) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at com.sun.jna.Library$Handler.invoke(Library.java:270) ~[jna-5.13.0.jar:5.13.0 (b0)]
        at jdk.proxy4/jdk.proxy4.$Proxy29.sysctlbyname(Unknown Source) ~[?:?]
        at oshi.util.platform.mac.SysctlUtil.sysctl(SysctlUtil.java:93) ~[oshi-core-6.4.13.jar:6.4.13]
        at org.opensearch.knn.jni.PlatformUtils.lambda$isAVX2SupportedBySystem$0(PlatformUtils.java:53) ~[opensearch-knn-3.0.0.0-SNAPSHOT.jar:3.0.0.0-SNAPSHOT]

Reply:
got it. thanks